core/src/main/java/org/apache/accumulo/core/clientImpl/TableOperationsImpl.java

```diff
@@ -1124,6 +1124,13 @@ private Map<String,String> modifyPropertiesUnwrapped(String tableName,
       Throwable cause = ae.getCause();
       if (cause instanceof TableNotFoundException) {
         throw new TableNotFoundException(null, tableName, null, ae);
+      } else if (cause instanceof ThriftTableOperationException) {
+        var ttoe = (ThriftTableOperationException) cause;
+        if (ttoe.getType() == TableOperationExceptionType.NAMESPACE_NOTFOUND) {
+          throw new TableNotFoundException(tableName, new NamespaceNotFoundException(ttoe));
+        } else if (ttoe.getType() == TableOperationExceptionType.NOTFOUND) {
+          throw new TableNotFoundException(ttoe);
+        }
```

Comment:
I would have expected that this would have been handled elsewhere. The ThriftTableOperationException shouldn't be directly wrapped by AccumuloException if it's a TNFE. Do you know where this is coming from? Where that wrapping is happening? It would probably be better to fix it closer to the cause.

Reply:
No, not sure where the wrapping is happening. I'll look into it

Reply:
Pushed a new commit 8a50b85b930dd623677850bb233d6d255263f28e
Please see the commit message there.
This pushes handling of ThriftTableOperationException closer to the thrift code instead of handling in the implementation of the API methods. Not sure how much better this is, but this avoids:
> The ThriftTableOperationException shouldn't be directly wrapped by AccumuloException if it's a TNFE

Reply:
If this change looks good, potentially other areas in TableOperationsImpl, NamespaceOperationsImpl, SecurityOperationsImpl, and/or InstanceOperationsImpl which handling of a thrift exception can be pushed closer to thrift code.